### ISSUE TITLE
Support for Azure Functions v4 (?)

### ISSUE THREAD

**Issue by django-speyk** [apollo-server-integrations/apollo-server-integration-azure-functions#41]
Is there a way to switch to V4 of the Azure Functions? The [docs](https://learn.microsoft.com/en-us/azure/azure-functions/functions-node-upgrade-v4?tabs=azure-cli-set-indexing-flag%2Cv4) say that you shouldn't have a ```function.json```, but it seems that that is necessary to use ```startServerAndCreateHandler```?

How should you use ```startServerAndCreateHandler``` in this no-json V4 model?

Thanks in advance!

**Comment by nnoce14** [comment]
Encountering the same issue as well. Seems the package is still using @azure/functions^3.2.0 when it should be using @azure/functions 4.x which changed how function apps are defined in nodejs

**Comment by aaronpowell** [comment]
This is related to #42.

Azure Functions v4 is a new programming model for JavaScript/TypeScript and it sees quite a few changes to the types that are used, so it's not a direct upgrade. As it's still in public preview I'm holding off supporting Azure Functions v4.

**Comment by gidich** [comment]
Makes sense, if you do plan on creating a beta version on a separate branch and are looking for community support I believe some of us on this thread may be interested in helping.

**Comment by aaronpowell** [comment]
I did explore it early on (when we were prototyping v4) to see what the approach might look like for building libraries on top of the new programming model. I really wanted to do:

```js
import { app } from "@azure/functions";

app.graphql(/* stuff here */);
```

But it's not possible and ended up having some discussion with the Functions team on what could be good options and there's a few to try but... well... time 😅

**Comment by connordoner** [comment]
Has any further progress been made on this issue, please? Happy (and eager!) to contribute but want to avoid duplicating anyone else's work. Thanks.

**Comment by aaronpowell** [comment]
No I haven't.

I've moved out of the JavaScript space at Microsoft so am rather time poor to tackle some of the stuff here.

**Comment by connordoner** [comment]
No worries, @AaronPowell! I'm happy to start pulling together a PR if you're willing/able to review it?

**Comment by aaronpowell** [comment]
> No worries, @aaronpowell! I'm happy to start pulling together a PR if you're willing/able to review it?

More than happy to review

**Comment by connordoner** [comment]
Thanks! I might take a little while as I have lots of other stuff on my plate right now and I'm only just getting started with Azure Functions, but I'll submit a proof of concept in the next month or so, and we can no doubt all work towards improving it. Appreciate your help.

**Comment by TengoCharlie** [comment]
@aaronpowell I have fixed the current support of V4 in the current library, Am I able to contribute the code??

**Comment by aaronpowell** [comment]
> @aaronpowell I have fixed the current support of V4 in the current library, Am I allowed to contribute the code??

Sure, submit a PR

**Comment by TengoCharlie** [comment]
Hello everyone, can someone help me set up the repository standard code with the solution?

**Comment by douglas-wang** [comment]
@aaronpowell  @TengoCharlie is there any progress on this please?
Azure functions v4 model is officially live now not in beta anymore.

**Comment by Infracloud-harsh** [comment]
@douglas-wang , am looking into it. Will raise a PR soon

**Comment by TengoCharlie** [comment]
> > @aaronpowell I have fixed the current support of V4 in the current library, Am I allowed to contribute the code??
> 
> Sure, submit a PR

I have changes the code as per the function V4. but i have a question that are we going to release another package for function v4 or we will provide support for v4 and v3 by same package?

**Comment by aaronpowell** [comment]
It'd be best to support them both in a single package

**Comment by barddoo** [comment]
@aaronpowell I just opened a PR to address this issue https://github.com/apollo-server-integrations/apollo-server-integration-azure-functions/pull/66

**Comment by danielRChildrens** [comment]
Thanks @barddoo, I have tested your PR and works perfectly with Azure functions v4.1.0 and Apollo server v4.10.0

**Comment by gidich** [comment]
@trevor-scheer - This repo has gone stale from an Apollo support standpoint (the last update was 3 months ago) - several people need to update to Azure Functions Programming model v4, and this pull request is blocking them - It looks like @aaronpowell has their attention turned toward other Microsoft priorities, is this something you can help move forward in his absence?

**Comment by charlesgardyn** [comment]
As a follow up on this, I just noticed that one cannot access the body in the get context function as one could in V3, as the body can only be parsed once.
https://github.com/apollo-server-integrations/apollo-server-integration-azure-functions/pull/77

**Comment by yasir-ajc458** [comment]
could you please give any working sample for
 `app.http('graphql', {
  handler: v4.startServerAndCreateHandler(server),
});`
I am getting below error:
`[2025-02-27T11:02:33.880Z] 1 functions found (Custom)
[2025-02-27T11:02:33.896Z] 0 functions loaded
[2025-02-27T11:02:33.916Z] Generating 0 job function(s)
[2025-02-27T11:02:33.943Z] No job functions found. Try making your job classes and methods public. If you're using binding extensions (e.g. Azure Storage, ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. builder.AddAzureStorage(), builder.AddServiceBus(), builder.AddTimers(), etc.).`

**Comment by connorgurney-personal** [comment]
Can you please share the rest of your function code, @yasir-ajc458?

**Comment by charlesgardyn** [comment]
You need this in your `package.json`
```
// ts
"main": "dist/src/functions/*.js"
// js
"main": "src/functions/*.js"
```

**Comment by charlesgardyn** [comment]
@yasir-ajc458
